src/binary-reader-objdump.c

```diff
@@ -366,16 +390,22 @@ static WasmResult on_import(uint32_t index,
                             WasmStringSlice module_name,
                             WasmStringSlice field_name,
                             void* user_data) {
-  print_details(user_data, " - " PRIstringslice " " PRIstringslice "\n",
-                  WASM_PRINTF_STRING_SLICE_ARG(module_name),
-                  WASM_PRINTF_STRING_SLICE_ARG(field_name));
+  Context* ctx = user_data;
+  ctx->import_module_name = module_name;
+  ctx->import_field_name = field_name;
   return WASM_OK;
 }
 
 static WasmResult on_import_func(uint32_t index,
                                  uint32_t sig_index,
                                  void* user_data) {
-  print_details(user_data, " - func sig=%d\n", sig_index);
+  Context* ctx = user_data;
+  print_details(
+      user_data, " - func[%d] sig=%d " PRIstringslice "." PRIstringslice "\n",
+      ctx->function_index, sig_index,
+      WASM_PRINTF_STRING_SLICE_ARG(ctx->import_module_name),
```

Comment:
Actually, the previous way is better; we should print the module and field names for all imported types, not just functions.